experimental/packages/opentelemetry-instrumentation/src/platform/node/instrumentation.ts

```diff
@@ -135,22 +141,30 @@ export abstract class InstrumentationBase<T = any>
     }
 
     for (const module of this._modules) {
-      this._hooks.push(
-        RequireInTheMiddle(
-          [module.name],
-          { internals: true },
-          (exports, name, baseDir) => {
-            return this._onRequire<typeof exports>(
-              (module as unknown) as InstrumentationModuleDefinition<
-                typeof exports
-              >,
-              exports,
-              name,
-              baseDir
-            );
-          }
-        )
-      );
+      // Note that for some reasons we can't share the hook callback, it mess up
+      // the context in which onRequireOrImport is called
```

Comment:
Just out of curiosity: what context is being messed up?

Reply:
I will precise in the comment but if the callback of `ESMHook` or `RequireInTheMiddle` is one function defined above and shared for both callbacks, the context within `_onRequireOrImport` isn't the instrumentation class (where `this._extractPackage` would not find _extractPackage function) but another one which i suspected come from ritm/iitm.